chain/network/src/network_protocol/propagator.rs

```diff
@@ -0,0 +1,75 @@
+use crate::network_protocol::proto::trace_context::SamplingPriority;
+use crate::network_protocol::proto::TraceContext;
+use opentelemetry::trace::{SpanContext, SpanId, TraceContextExt, TraceFlags, TraceId, TraceState};
+use opentelemetry::Context;
+use protobuf::{EnumOrUnknown, MessageField};
+
+const TRACE_FLAG_DEFERRED: TraceFlags = TraceFlags::new(0x02);
+
+#[derive(Debug)]
+pub(crate) enum ExtractError {
+    TraceId,
+    SpanId,
+}
+
+#[derive(Debug, Default, Clone)]
+pub(crate) struct NodePropagator {}
```

Comment:
document what's the purpose of this struct.

Reply:
do we really need this type in the first place? Wouldn't a bunch of standalone functions do equally well?